### PR TITLE
nheko, use bundled mtxclient for now (static), undefined references w…

### PR DESCRIPTION
**Pull request by Begasus**
…ith new nlohmann_json

### CODE REVIEW

**Comment by Begasus** [comment]
New nlohmann_json is needed for newer (upcomming) nheko so not reverting that, using the bundled mtxclient fixes the build, should be revised on the new version release.

File: net-im/nheko/nheko-0.11.3.recipe [hunk 1]
```diff
@@ -113,4 +119,6 @@ INSTALL()
 
 	addResourcesToBinaries nheko.rdef $appsDir/Nheko
 	addAppDeskbarSymlink $appsDir/Nheko
+
+	prepareInstalledDevelLib libmatrix_client
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:123]: What's the benefit? The static libs shouldn't be needed at runtime.

Comment (Begasus) [reply net-im/nheko/nheko-0.11.3.recipe:123]: It comes included as bundled mtxclient to fix the build, should be fine to remove on a version bump for Nheko

Comment (Begasus) [reply net-im/nheko/nheko-0.11.3.recipe:123]: I could run another check to see if it launches fine building with the bundled one and using our shared library in the mtxclient package.

Comment (korli) [reply net-im/nheko/nheko-0.11.3.recipe:123]: I mean, the headers and the libs shouldn't be in the mheko package. Please remove them.

Comment (Begasus) [reply net-im/nheko/nheko-0.11.3.recipe:123]: Was looking into that when checking earlier, should be resolved in https://github.com/haikuports/haikuports/pull/9776

**Pull request merged**